feat: Add an example of generating openapi spec using tools pattern

hiddenmarten opened the pull request:
Hello everyone!

I'm new to the community, and I couldn't find an easy way to generate an OpenAPI spec during the build process. So, I came up with a straightforward solution that might be useful to others.

P.S. I use the generated OpenAPI spec to create clients for testing purposes.

P.P.S. If there is a way to generate it using cli or smth like that, I would be very happy. :)

EwenQuim commented:
Hi @hiddenmarten ! Thanks for this PR 🎉  That will be really useful for future developers.

I get what you're trying to show but i think there are other solutions than what you proposed. Here are 2 ways I'll do that:

## Solution 1

From inside with environment variables. This might feel a bit like "side-effects" but it works well.

<img width="483" alt="image" src="https://github.com/user-attachments/assets/ba6dda37-3e24-4e49-b8c5-4d696d4681b6" />


## Solution 2

Via different Engine `options` in each case.

<img width="491" alt="image" src="https://github.com/user-attachments/assets/bef27ec5-5b5f-47ee-b3bc-859ac707666d" />

- in `main.go` to run the server without spec:

<img width="442" alt="image" src="https://github.com/user-attachments/assets/65ab4784-5d2b-4196-8b8a-faa91d34f5f9" />


- in `genspec.go` to generate the spec without running the server:

<img width="1136" alt="image" src="https://github.com/user-attachments/assets/5e3808de-2cb5-4136-a44c-e78b59589767" />

hiddenmarten commented:
Hi, @EwenQuim!

Thanks for the rapid answer!

I'll check your suggestions a bit later, really appreciate your help.

EwenQuim commented:
I'll love it if you can change your PR so the best of these solution are available to the general public :)

hiddenmarten commented:
I've updated the example. 

I implemented it slightly differently than what was suggested, but I believe it still conveys the same idea. I intentionally left GetServer() without any parameters because I consider it a "central" configuration that should remain unchanged once set up. I see this implementation as a bit simpler example, especially for the newcomers.

If this topic is really important to you, I'll modify it to match your proposed solution as closely as possible.

EwenQuim commented:
Really good idea, thank you!

EwenQuim commented:
Thanks for your contribution 🎉
